app/models/studio.rb

```diff
@@ -1,3 +1,9 @@
 class Studio < ApplicationRecord
+  validates :name, presence: true
+  validates :location, presence: true
   has_many :movies
+
+  def actors
+    Actor.joins(movies: :studio).where("studios.id = ?", id).distinct.pluck(:name)
```

Comment:
Nice AR! This query is fine, but we can write an even shorter query by relying on associations. Check this out: 

If we add `has_many :actors, through: :movies` to this Studio model, then our query becomes
```ruby
actors.distinct.pluck(:name)
```

The reason this works is because a Movie already `has_many :actors`, so thanks to some transitive logic, a Studio also has many Actors through its Movies. 

Both ways of writing this query works!